drivers/filesystems/fastfat/volume.c

```diff
@@ -100,13 +100,14 @@ FsdGetFsAttributeInformation(
     ASSERT(*BufferLength >= sizeof(FILE_FS_ATTRIBUTE_INFORMATION));
     *BufferLength -= FIELD_OFFSET(FILE_FS_ATTRIBUTE_INFORMATION, FileSystemName);
 
-    if (DeviceExt->FatInfo.FatType == FAT32)
+    switch (DeviceExt->FatInfo.FatType)
     {
-        pName = L"FAT32";
-    }
-    else
-    {
-        pName = L"FAT";
+        case FAT12: pName = L"FAT"; break;
+        case FAT16: pName = L"FAT"; break;
+        case FAT32: pName = L"FAT32"; break;
```

Comment:
Would this driver load for regular FAT partitions as well ? Maybe we should prefer the other one.

Reply:
`FsRecIsFatxVolume` only accepts `FATX` partitions

Reply:
Speaking of filesystem recognition, I wonder... how Windows implements it? Is our implementation different from it? Because for example I'm pretty sure Windows does not have BTRFS filesystem detection code, however it's possible to use that filesystem by just supplying the FS driver.

Reply:
A FS recognizer is just a FS (declares as such with the kernel), it's just that it's super-lightweight and when a suitable FS is detected by it, it asks the kernel to load the actual FS driver.
If BTRFS works obviously on Windows (and Windows' fs_rec doesn't have BTRFS support), it's just because either the fullblown BTRFS driver is always loaded, or, because WinBtrfs also includes a small BTRFS-specific recognizer (that would also be a FS).

Reply:
So, in ReactOS' case, instead of having also a separate smaller BTRFS driver whose purpose only would be to recognize Btrfs volumes and load the fullfledged BTRFS driver, we instead incorporate BTRFS detection within our single fs_rec.